packages/web/src/components/home/card-list/CardList.tsx

```diff
@@ -2,44 +2,69 @@ import IconTriangleArrowUp from "@components/common/icons/IconTriangleArrowUp";
 import IconTriangleArrowDown from "@components/common/icons/IconTriangleArrowDown";
 import { CardListWrapper, ListItem } from "./CardList.styles";
 import DoubleLogo from "@components/common/double-logo/DoubleLogo";
-import { CardListPoolInfo, CardListTokenInfo } from "@models/common/card-list-item-info";
+import {
+  CardListKeyStats,
+  CardListPoolInfo,
+  CardListTokenInfo,
+} from "@models/common/card-list-item-info";
 import { useCallback, useMemo } from "react";
 import { SwapFeeTierInfoMap } from "@constants/option.constant";
 import MissingLogo from "@components/common/missing-logo/MissingLogo";
 
 interface CardListProps {
-  list: Array<CardListTokenInfo | CardListPoolInfo>;
+  list: Array<CardListTokenInfo | CardListPoolInfo | CardListKeyStats>;
   onClickItem: (id: string) => void;
   isHiddenIndex?: boolean;
 }
 
-function isTokenInfo(info: CardListTokenInfo | CardListPoolInfo): info is CardListTokenInfo {
+const enum CardType {
+  "keyStats",
+  "token",
+  "pool",
+}
+
+function typeTokenInfo(
+  info: CardListTokenInfo | CardListPoolInfo | CardListKeyStats,
+): CardType {
   if ("token" in info) {
-    return true;
+    return CardType.token;
+  }
+  if ("label" in info) {
+    return CardType.keyStats;
   }
-  return false;
+  return CardType.pool;
 }
 
-const CardList: React.FC<CardListProps> = ({ list, onClickItem, isHiddenIndex = false }) => {
-
+const CardList: React.FC<CardListProps> = ({
+  list,
+  onClickItem,
+  isHiddenIndex = false,
+}) => {
   return (
     <CardListWrapper>
       {list.map((item, index) =>
-        isTokenInfo(item) ? (
+        typeTokenInfo(item) === CardType.token ? (
           <CardListTokenItem
             key={index}
             index={index + 1}
-            item={item}
+            item={item as CardListTokenInfo}
```

Comment:
Why did you remove type guard and use type assertion?
https://github.com/gnoswap-labs/gnoswap-interface/pull/316/files#diff-d9c968003505678c20746c58eeccb26e94d3c70a0d51758a806051405b80302cL16

Reply:
now we have 3 option for this

Reply:
You can declare a Wrapper component like this.
The Wrapper component just handles branching by type.

It seems better to explicitly branch by typeguard rather than type assertion (I think it's a matter of branching policy management).

#### CardList Component
```
const CardList: React.FC<CardListProps> = ({
  list,
  onClickItem,
  isHiddenIndex = false,
}) => {
  return (
    <CardListWrapper>
      {list.map((item, index) => (
        <CardListItem
          key={index}
          index={index + 1}
          item={item}
          onClickItem={onClickItem}
          isHiddenIndex={isHiddenIndex}
        />
      ))}
    </CardListWrapper>
  );
};
```

#### CardListItem Component
```
const CardListItem: React.FC<CardListItemProps> = ({
  index,
  item,
  isHiddenIndex,
  onClickItem,
}) => {
  function isTokenCardItem(
    item: CardListTokenInfo | CardListPoolInfo | CardListKeyStats,
  ): item is CardListTokenInfo {
    return "token" in item;
  }

  function isKeyStatsCardItem(
    item: CardListTokenInfo | CardListPoolInfo | CardListKeyStats,
  ): item is CardListKeyStats {
    return "label" in item;
  }

  if (isTokenCardItem(item)) {
    return (
      <CardListTokenItem
        index={index}
        item={item}
        isHiddenIndex={isHiddenIndex}
        onClickItem={onClickItem}
      />
    );
  }

  if (isKeyStatsCardItem(item)) {
    return <CardListKeyStatsItem item={item} />;
  }

  return (
    <CardListPoolItem index={index} item={item} onClickItem={onClickItem} />
  );
};
```

Reply:
Done